src/mod_muc.erl

```diff
@@ -510,20 +505,28 @@ process_packet(Acc, From, To, El, #state{
 route_to_room(<<>>, {_, To, _Acc, _} = Routed, State) ->
     {_, _, Nick} = jid:to_lower(To),
     route_by_nick(Nick, Routed, State);
-route_to_room(Room, {From, To, Acc, Packet} = Routed, #state{host=Host} = State) ->
+route_to_room(Room, Routed, #state{host=Host} = State) ->
     case mnesia:dirty_read(muc_online_room, {Room, Host}) of
         [] ->
-            route_to_nonexistent_room(Room, Routed, State);
+            case route_to_nonexistent_room(Room, Routed, State) of
+                {ok, NewPid} ->
+                    route_to_online_room(NewPid, Routed);
+                {exists, OldPid} ->
+                    route_to_online_room(OldPid, Routed)
+            end;
         [R] ->
             Pid = R#muc_online_room.pid,
-            ?DEBUG("MUC: send to process ~p~n", [Pid]),
-            {_, _, Nick} = jid:to_lower(To),
-            mod_muc_room:route(Pid, From, Nick, Acc, Packet),
-            ok
+            route_to_online_room(Pid, Routed)
     end.
 
+route_to_online_room(Pid, Routed) ->
+    ?DEBUG("MUC: send to process ~p~n", [Pid]),
+    {From, To, Acc, Packet} = Routed,
+    {_, _, Nick} = jid:to_lower(To),
+    mod_muc_room:route(Pid, From, Nick, Acc, Packet),
```

Comment:
I know that this is how it was, but it doesn't seem right to ignore the return value of this function. Maybe we should at least match it, even if the spec says it's always 'ok'. What do you think about it?